chapters/ch04.2-writing-logs.md

```diff
@@ -411,6 +411,13 @@ class Logger {
 
     async init() {
         const file_name = this.#config.file_prefix + new Date().toISOString().replace(/\..+/, "") + ".log";
+
+        /**Expcted filename looks like this -> "LogTar_2023-08-26T14:46:11.log" this works well with Linux
+        * but it thorws an error if it is Windows because windows doesn't allow the colon ":" in filename
+        * So we could write this line an replace the dots and clolons with a -
+        * const file_name = const file_name=this.#config.file_prefix+ new Date().toISOString().replace(/[\..+::+]/,"-")+".log";
+        * Expcted filename looks like this -> "LogTar_2023-08-26T14-46-11.log" 
+        */
```

Comment:
We can remove the comment probably, and simplify the regex to be `[\.:]+`
So, the line
```js
const file_name = this.#config.file_prefix + new Date().toISOString().replace(/\..+/, "") + ".log";
```
becomes
```js
const file_name = this.#config.file_prefix + new Date().toISOString().replace(/[\.:]+/, "-") + ".log";
```
I'll do the rest, i.e add an extra line explaining about the `:` in the  - [A small primer to regular expressions](https://github.com/ishtms/learn-nodejs-hard-way/blob/master/chapters/ch04.2-writing-logs.md#a-small-primer-on-regular-expressions) section as well as explaining the `[]` regex operator.

Reply:
Thank you for your feedback and guidance, @ishtms . I'm fully prepared to address the changes you've highlighted in the comments